cmd/add.go

```diff
@@ -1,54 +1,155 @@
 	}
+
 	tags, err := cmd.Flags().GetStringArray("tag")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get tag flag: %w", err)
 	}
+
 	filename, err := cmd.Flags().GetString("filename")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get filename flag: %w", err)
+	}
+
+	cache, err := cmd.Flags().GetString("cache")
+	if err != nil {
+		return fmt.Errorf("failed to get cache flag: %w", err)
+	}
+
+	// Validate cache configuration
+	if cache != "" {
+		// If cache URL is provided, authentication is required
+		token := os.Getenv("ARTIFACTORY_TOKEN")
+		if token == "" {
+			return fmt.Errorf("ARTIFACTORY_TOKEN must be set when using cache")
+		}
+		log.Debug().
+			Str("cache_url", cache).
+			Msg("Artifactory cache configured")
+	}
+
+	// Create or open lock file
+	lock, err := internal.NewLock(lockFile, true)
+	if err != nil {
+		return fmt.Errorf("failed to create/open lock file: %w", err)
+	}
+
+	// Check if resource is local file
+	isLocal := !strings.HasPrefix(args[0], "http://") && !strings.HasPrefix(args[0], "https://")
+	if isLocal && cache != "" {
+		log.Debug().
+			Str("file", args[0]).
+			Str("cache", cache).
+			Msg("Uploading local file to Artifactory")
+
+		if err := uploadToArtifactory(args[0], cache); err != nil {
+			return fmt.Errorf("failed to upload to Artifactory: %w", err)
+		}
+	}
+
+	// Add resource to lock file
+	if err := lock.AddResourceWithCache(args, algo, tags, filename, cache); err != nil {
+		return fmt.Errorf("failed to add resource: %w", err)
+	}
+
+	// Save changes
+	if err := lock.Save(); err != nil {
+		return fmt.Errorf("failed to save lock file: %w", err)
+	}
+
+	log.Debug().
+		Str("url", args[0]).
+		Str("cache", cache).
+		Msg("Resource added successfully")
+
+	return nil
+}
+
+// uploadToArtifactory uploads a file to Artifactory
+func uploadToArtifactory(filePath, cacheUrl string) error {
+	// Get authentication token
+	token := os.Getenv("ARTIFACTORY_TOKEN")
+	if token == "" {
+		return fmt.Errorf("ARTIFACTORY_TOKEN is not set")
+	}
+
+	// Read file
+	fileData, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to read file: %w", err)
 	}
-	err = lock.AddResource(args, algo, tags, filename)
+
+	// Create request
+	req, err := http.NewRequest(http.MethodPut, cacheUrl, bytes.NewReader(fileData))
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create request: %w", err)
 	}
-	err = lock.Save()
+
+	// Set headers
+	req.Header.Set("Authorization", "Bearer "+token)
+	req.Header.Set("Content-Type", "application/octet-stream")
+	req.Header.Set("Content-Length", fmt.Sprintf("%d", len(fileData)))
+
+	// Send request
+	client := &http.Client{}
+	resp, err := client.Do(req)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to execute request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Check response
+	if resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusOK {
```

Comment:
I'm not sure the logic test is right here.

Don't both of these need to be true?